resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/internal/CircuitBreakerStateMachine.java

```diff
@@ -695,12 +706,23 @@ public CircuitBreakerMetrics getMetrics() {
             return circuitBreakerMetrics;
         }
 
+        @Override
+        public void preTransitionHook() {
+            cancelAutomaticTransitionToHalfOpen();
+        }
+
         private void toHalfOpenState() {
             if (isOpen.compareAndSet(true, false)) {
                 transitionToHalfOpenState();
             }
         }
 
+        private void cancelAutomaticTransitionToHalfOpen() {
+            if (transitionToHalfOpenFuture != null) {
+                transitionToHalfOpenFuture.cancel(true);
```

Comment:
what happens if the future is already completed?

Reply:
does it really work? 
https://stackoverflow.com/questions/29181905/future-cancel-is-not-cancelling-the-scheduled-execution-of-scheduledexecutorse

Reply:
Yes. Needed to check whether it is done or is cancelled already.
Thank you.